Adding Variable Fonts Hook update (#26)
aenhancers/after-effects-scripting-guide: docs/other/fontobject.rst

Attributes
==========

.. _FontObject.designAxesData:

FontObject.designAxesData
*********************************************

``app.fonts.allFonts[0][0].designAxesData``

**Description**

> Retruns an Array of Objects, containing the design axes data from the font. 

Reviewer: "Retruns" should be "Returns"

> Each object is composed of the axis name, tag, min value and max value.

Reviewer 2: Will return undefined for non variable fonts.

**Example**

This example will select the first returned Font Family Array.

.. code:: javascript

   // Getting the first available Variable Font on the system
   var firstVariableFont = fontsWithDefaultDesignAxes()[0];
   var axesData = firstVariableFont.designAxesData;

   // Getting the first design axis for that Font 
   var firstAxis = axesData[0];

   alert(firstAxis.name+"\n"+firstAxis.tag+"\n"+firstAxis.min+"\n"+firstAxis.max);

**Type**

Array of Objects; read-only.

----

.. _FontObject.designVector:

FontObject.designVector
*********************************************

``app.fonts.fontsWithDefaultDesignAxes()[0].designVector``

**Description**

> For Variable fonts will return an ordered array of length matching the number of design axes defined by the font”

Reviewer 2: trailing "

> For Variable fonts will return an ordered array of length matching the number of design axes defined by the font”

Reviewer: Change preposition and remove errant quotation mark.
```suggestion
For Variable fonts will return an ordered array with a length matching the number of design axes defined by the font.
```

.. note::

> Will return undefined for non variable fonts. 

Reviewer: Add a tab character at the start of this line to avoid a build warning.

Reviewer: Add hyphen here:
```suggestion
Will return undefined for non-variable fonts. 
```

**Type**

> Array of Floats; read-only.

Reviewer: ```suggestion
Array of floating-point values; read-only.
```

----

.. _FontObject.familyPrefix:

FontObject.familyPrefix
*********************************************

``app.fonts.fontsWithDefaultDesignAxes()[0].familyPrefix``

**Description**

> The family prefix of the variable font.

Reviewer 2: For example, the family of the PostScript name "SFPro-Bold" is "SFPro".

Also, will return undefined for non variable fonts.

-  ``CTScript.CT_TAILE_SCRIPT``
-  ``CTScript.CT_TAIVIET_SCRIPT``
-  ``CTScript.CT_DONTKNOW_SCRIPT``

=======
Methods
=======

.. _FontObject.hasSameDict:

FontObject.hasSameDict()
**********************************************

> ``app.fonts.fontsWithDefaultDesignAxes()[0].hasSameDict(Font)``

Reviewer: ```suggestion
``app.fonts.fontsWithDefaultDesignAxes()[0].hasSameDict(fontObject)``
```

**Description**

> This function will true if the Font Object passed as an argument share the same variable font dictionnary.

Reviewer 2: spelling - dictionary

"shares"

> This function will true if the Font Object passed as an argument share the same variable font dictionnary.

Reviewer: ```suggestion
This function will return true if the Font Object passed as an argument shares the same variable font dictionary as the Font object the function is called on.
```

Reviewer 3: Can this clarify what "dictionary" means in this context? And when you would want to use this.

Reviewer 2: Font files are in one sense simply a "dictionary" which maps from characters to glyphs.

You have the character "A", you ask the font for glyph(s) which map to that character.

It is a bit more complicated than that but it is a good paradigm.

Variable fonts in particular mess with the terms because here one _physical_ font file on disk (aka a font dictionary) supports _many_ different font instances/faces.

This leads to all sorts of language problems with these new Font scripting objects since comparing two different Font object instances and asking them if they are the same..._what do you mean by same?_

With this API we are simply providing a way to compare to Font objects to see if they are mapped in some way to the same physical font file (dictionary) on disk.

.. note::
  Can only return true when used on variable fonts with the argument being a Font Object of a variable font.

**Parameters**

====================  ========================================================

> Font                  A Font Object

Reviewer: ```suggestion
fontObject                  A Font Object
```

====================  ========================================================

**Returns**

A Boolean.

----

.. _FontObject.postScriptNameForDesignVector:

FontObject.postScriptNameForDesignVector()
**********************************************

> ``app.fonts.fontsWithDefaultDesignAxes()[0].postScriptNameForDesignVector([Design Vector])``

Reviewer: ```suggestion
``app.fonts.fontsWithDefaultDesignAxes()[0].postScriptNameForDesignVector([...vectorValues])``
```

**Description**

This function will return the postscript name of the variable font for the specific design vectors passed as the argument.

**Parameters**

====================  ========================================================

> Design Vector          An array of float values.

Reviewer: ```suggestion
vectorValues          An array of float values that matches the length of :ref:`FontObject.designVector` for the given variable font.
```
